core/vcpu.c

```diff
@@ -1079,6 +1079,24 @@ static void load_dirty_vmcs_fields(struct vcpu_t *vcpu)
     }
 
     // interruptibility
+    // 26.3.1.5 Checks on Guest Non-Register State
+    // Bit 0 (blocking by STI) must be 0 if the IF flag (bit 9) is 0 in the
+    //   RFLAGS field.
+    // This is a WA to fix the VM-entry failure due to invalid guest state,
+    //   sometimes when a snapshot is loaded but IF and interruptibility_state
+    //   don't pass the checks as mentioned in SDM 26.3.1.5.
+    // In in-order execution, interruptibility_state is updated when advancing
+    //   the IP. However when a snapshot is loaded, EFLAGS are restored but
+    //   guest non-register state not restored.
+    //   TODO: Find better approach instead of letting the check pass.
+    if (!(state->_rflags & EFLAGS_IF)) {
+        if (vmx(vcpu, interruptibility_state).raw &
+            GUEST_INTRSTAT_STI_BLOCKING) {
+            vmx(vcpu, interruptibility_state).raw &=
+                ~GUEST_INTRSTAT_STI_BLOCKING;
```

Comment:
Use 8-space indent for long line break, while the 4-space indent is for normal line wrap.